src/imio/directory/core/contents/entity/export.py

```diff
@@ -0,0 +1,211 @@
+    "mails",
+    "phones",
+    "urls",
+    "private_mails",
+    "private_phones",
+    "private_urls",
+    "private_note",
+    "vat_number",
+    "schedule",
+    "multi_schedule",
+    "schedule.monday.morningstart",
+    "schedule.monday.morningend",
+    "schedule.monday.afternoonstart",
+    "schedule.monday.afternoonend",
+    "schedule.monday.comment",
+    "schedule.tuesday.morningstart",
+    "schedule.tuesday.morningend",
+    "schedule.tuesday.afternoonstart",
+    "schedule.tuesday.afternoonend",
+    "schedule.tuesday.comment",
+    "schedule.wednesday.morningstart",
+    "schedule.wednesday.morningend",
+    "schedule.wednesday.afternoonstart",
+    "schedule.wednesday.afternoonend",
+    "schedule.wednesday.comment",
+    "schedule.thursday.morningstart",
+    "schedule.thursday.morningend",
+    "schedule.thursday.afternoonstart",
+    "schedule.thursday.afternoonend",
+    "schedule.thursday.comment",
+    "schedule.friday.morningstart",
+    "schedule.friday.morningend",
+    "schedule.friday.afternoonstart",
+    "schedule.friday.afternoonend",
+    "schedule.friday.comment",
+    "schedule.saturday.morningstart",
+    "schedule.saturday.morningend",
+    "schedule.saturday.afternoonstart",
+    "schedule.saturday.afternoonend",
+    "schedule.saturday.comment",
+    "schedule.sunday.morningstart",
+    "schedule.sunday.morningend",
+    "schedule.sunday.afternoonstart",
+    "schedule.sunday.afternoonend",
+    "schedule.sunday.comment",
+    "exceptional_closure",
+    "type",
+    "facilities",
+    "iam",
+    "taxonomy_contact_category",
+    "topics",
+    "subject",
+    "modified",
+]
+
+
+class ExportView(BrowserView):
+
+    def __call__(self):
+        self.lang = api.portal.get_current_language()
+        entity_id = self.context.id
+        dict_datas = self.get_directory_response(entity_id)
+        items = dict_datas.get("items", None)
+        if not items:
+            return self.request.response.setStatus(204)
+        df = pandas.json_normalize(items)
+        dataframe_columns = list(df)
+        set1 = set(ordered_signifiant_columns)
+        set2 = set(dataframe_columns)
+        intersection = set1.intersection(set2)
+        columns = sorted(
+            intersection, key=lambda x: ordered_signifiant_columns.index(x)
+        )
+        df = df[columns]
+
+        csv_buffer = io.BytesIO()
+        df.to_csv(csv_buffer, index=False, encoding="utf-8")
+
+        self.request.response.setHeader(
+            "Content-Disposition", 'attachment; filename="export.csv"'
+        )
+        self.request.response.setHeader("Content-Type", "text/csv")
+        self.request.response.write(csv_buffer.getvalue())
+        return self.request.response
+
+    def get_directory_response(self, entity_id):
+        context_path = "/".join(self.context.getPhysicalPath())
+        query = {
+            "portal_type": "imio.directory.Contact",
+            "path": {"query": context_path, "depth": 1},
```

Comment:
Why use a `depth` of 1 ?  We could have contacts in contacts right ?

Reply:
Well seen ! 
Set to -1